doc/nrf/releases/release-notes-2.4.0.rst

```diff
@@ -0,0 +1,1038 @@
+
+  * Added:
+
+    * Sending of log messages directly to nRF Cloud.
+    * Overlay to enable `Zephyr Logging`_ backend for full logging to nRF Cloud.
+
+* :ref:`nrf_cloud_rest_fota` sample:
+
+    * Updated the sample by enabling external flash in the nRF9160 DK devicetree overlays for v0.14.0 or later versions, as it is now disabled in the Zephyr board definition.
+
+* :ref:`nrf_cloud_rest_device_message` sample:
+
+  * Added:
+
+    * Overlays to use RTT instead of UART for testing purposes.
+    * Sending of log messages directly to nRF Cloud.
+    * Overlay to enable `Zephyr Logging`_ backend for full logging to nRF Cloud.
+
+  * Updated the Hello World message sent to nRF Cloud, which now contains a timestamp (message ID).
+
+* :ref:`memfault_sample` sample:
+
+ * Added:
+
+    * Support for the nRF7002 DK.
+    * A Kconfig fragment to enable ETB trace.
+
+  * Updated:
+
+    * The sample has been moved from :file:`nrf9160/memfault` to :file:`debug/memfault`.
+      The documentation is now found in the :ref:`debug_samples` section.
+
+* :ref:`modem_trace_flash` sample:
+
+  * Updated the sample by enabling external flash in the nRF9160 DK devicetree overlays for v0.14.0 or later versions, as it is now disabled in the Zephyr board definition.
+
+Trusted Firmware-M (TF-M) samples
+---------------------------------
+
+* :ref:`provisioning_image` sample:
+
+  * Updated by moving the network core logic to the new sample :ref:`provisioning_image_net_core` instead of being a Zephyr module.
+
+Thread samples
+--------------
+
+|no_changes_yet_note|
+
+Matter samples
+--------------
+
+* Updated the default settings partition size for all Matter samples from 16 kB to 32 kB.
+
+  .. caution::
+      This change can affect the Device Firmware Update (DFU) from the older firmware versions that were using the 16-kB settings size.
+      Read more about this in the :ref:`ug_matter_device_bootloader_partition_layout` section of the Matter documentation.
+      You can still perform DFU from the older firmware version to the latest firmware version, but you will have to change the default settings size from 32 kB to the value used in the older version.
+
+* :ref:`matter_lock_sample` sample:
+
+  * Added the Matter Nordic UART Service (NUS) feature, which allows controlling the door lock device remotely through Bluetooth LE using two simple commands: ``Lock`` and ``Unlock``.
+    This feature is dedicated for the nRF52840 and the nRF5340 DKs.
+    The sample supports one Bluetooth LE connection at a time.
+    Matter commissioning, DFU, and NUS over Bluetooth LE must be run separately.
+  * Updated the default value of the feature map.
+
+* :ref:`matter_light_bulb_sample` sample:
+
+  * Added handling of TriggerEffect command.
+
+* :ref:`matter_window_covering_sample` sample:
+
+  * Updated by aligning default values of Window Covering attributes with the specification.
+
```

Comment:
Can we still add one more generic comment in Matter samples section:


Door Lock, Window Covering, Light Bulb and Weather Station applications have been tested with the following ecosystems:
 - Google Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions. Tested with Google Nest Hub 2nd generation (software version: 47.9.4.447810048; Chromecast firmware version: 1.56.324896, and Google Home mobile application v3.0.1.9).
 - Apple Home ecosystem for both Matter over Thread and Matter over Wi-Fi solutions. Tested with Apple HomePod mini and Apple iPhone (iOS v16.5).
 
Additionally, mentioned samples were positively verified against “Works with Google” certification tests.